pallets/parachain-staking/src/lib.rs

```diff
@@ -438,6 +442,31 @@ pub mod pallet {
 				weight = weight.saturating_add(T::OnNewRound::on_new_round(round.current));
 				// pay all stakers for T::RewardPaymentDelay rounds ago
 				weight = weight.saturating_add(Self::prepare_staking_payouts(round.current));
+
+				// select total candidates
+				let candidates = <CandidatePool<T>>::get().0;
+
+				// if candidates length is below or eq to 66% of max_collators,
+				// we don't mark any other collator as offline
+				let max_collators = <TotalSelected<T>>::get() as f32;
+
+				// iter candidates to check which of them must be marked as offline
+				for candidate in candidates.clone() {
+					if let Some(last_round) = <CandidateLastActive<T>>::get(&candidate.owner) {
+						if round.current.saturating_sub(last_round) > T::MaxOfflineRounds::get()
+							&& candidates.len() > (max_collators * 0.66) as usize
```

Comment:
Is it ok to use floating point in the runtime? If not, try something like this but with saturating logic:

```suggestion
							&& candidates.len() * 3 > (max_collators * 2) as usize
```